pkg/ops/iso.go

```diff
@@ -31,10 +38,38 @@ func GenISO(name, src, dst string, i schema.ISO) func(ctx context.Context) error
 		}
 
 		log.Info().Msgf("Generating iso '%s' from '%s' to '%s'", name, src, dst)
-		out, err := utils.SH(fmt.Sprintf("/entrypoint.sh --debug --name %s build-iso --squash-no-compression --overlay-iso %s --date=false --output %s dir:%s", name, overlay, dst, src))
-		log.Printf("Output '%s'", out)
+		cfg, err := enkiconfig.ReadConfigBuild("/config", &pflag.FlagSet{})
+		if err != nil {
+			return err
+		}
+		cfg.Name = name
+		cfg.OutDir = dst
+		cfg.Logger = sdkTypes.NewKairosLogger("enki", "debug", false)
+		imgSource := v1.NewDirSrc(src)
+		grub := v1.NewDirSrc("/grub2")
+		isoOverlay := v1.NewDirSrc(overlay)
+
+		// Removed here: https://github.com/kairos-io/osbuilder/commit/866dc42c48878fa8bebab26d8f24bf0f2953eb6e#diff-5e18c6520d9fb094d1d4c52b4fb652f09becd65053d95f8d7ac89c7998172fd4
+		// enki is called here: https://github.com/kairos-io/AuroraBoot/blob/9ca30fc6e3b6cac2d227e7937bd6895198bdc4d0/pkg/ops/iso.go#L34
+		// config is loaded here: https://github.com/kairos-io/enki/blob/844bd15261e89f103c16d1caa656ec41524b8d4f/pkg/config/config.go#L110-L114
+		// config was copied in place here: https://github.com/kairos-io/osbuilder/blob/405eda716a6c291eb539a98765a291e893d0eda1/tools-image/Dockerfile#L86
+		// which is part of the auroraboot image: https://github.com/kairos-io/AuroraBoot/blob/9ca30fc6e3b6cac2d227e7937bd6895198bdc4d0/Dockerfile#L1
+		// TODO: Cleanup this madness!
+
+		//uefi := v1.NewDirSrc("/efi")
+		//spec.UEFI = append(spec.UEFI, uefi)
```

Comment:
`/efi` and `/grub2` were populated here: https://github.com/kairos-io/osbuilder/blob/95509370f6a87229879f1a381afa5d47225ce12d/tools-image/Dockerfile#L29-L30

I skipped `/efi` and it still builds a bootable iso (in qemu at least) but it's not the case for `/grub2`. It's still needed. Are we sure the contents of `/efi` are now found in the rootfs image? @Itxaka any link on where this happens?

There are just 2 luet packages and luet is a golang binary. Would it be better if I just installed those 2 packages here instead of relying on the Dockerfile of the tools image to do that?

A better approach is to make the arguments of this command and allow the caller to set those directories. After all, it's the caller that knows where they installed those luet packages. And if those directories are required, we make them required arguments.

Reply:
copyShim and copyGrub both search in the rootfs for the files I think:

https://github.com/kairos-io/enki/blob/main/pkg/action/build-iso.go#L256
```
// copyShim copies the shim files into the EFI partition
// tempdir is the temp dir where the EFI image is generated from
// rootdir is the rootfs where the shim files are searched for
```
The only part in which we would fail is on alpine which doesnt ship signed efi files so we use our packaged ones:

```
		// All failed...maybe we are on alpine which doesnt provide shim/grub.efi ?
		// In that case, we can just use the luet packaged artifacts
```

In which case, we could provably just default to the ones they ship, after all alpine wont work with secureboot due to this, so we migth as well ship their own provided files and drop the luet packages once and for all.

Let me check the grub2 stuff

Reply:
yeah grub seems to do the same things, gets a list of possible grub2 efi files, then searches the rootfs for them to copy them into the EFI image for the iso... maybe the rootfs its searching it for its wrong?

Reply:
For now, I added them both back and a comment to indicate where they are coming from. Let's merge it as it is and clean up more in a future iteration.

Reply:
I think rootfs sh ould provde all those files, we cannot go backwards and let the user provide those, we already went that route and it was a mess, secureboot broken, different paths for everything and such. Distros already provide all the needed files and in the case of needing a generic file like eltorito.img, we could even ship a generic one bundled with auroraboot as its generic enough to work for everything.

Reply:
`copyEfi` and `copyGrub` both copy files from `/efi` . Nothing copies files [from `/grub2`](https://github.com/kairos-io/osbuilder/blob/95509370f6a87229879f1a381afa5d47225ce12d/tools-image/Dockerfile#L29).
I guess I can remove `/efi` at least from here.